### PR TITLE
Updating java version from 8 to 11

### PR DESCRIPTION
**Pull request by himanshugoyal1065**
### Closes Issue(s)

Closes #

### More

- [ ] Added/updated documentation

### CODE REVIEW

File: akka-bbb-fsesl/build.sbt [hunk 1]
```diff
@@ -28,8 +28,8 @@ val compileSettings = Seq(
 )
 
 resolvers ++= Seq(
-  "spray repo" at "http://repo.spray.io/",
-  "blindside-repos" at "http://blindside.googlecode.com/svn/repository/"
+  ("spray repo" at "http://repo.spray.io/").withAllowInsecureProtocol(true),
+  ("blindside-repos" at "http://blindside.googlecode.com/svn/repository/").withAllowInsecureProtocol(true)
```
Comment (basisbit) [inline akka-bbb-fsesl/build.sbt:32]: this repo seems to be an old / not existing repo? Also, please do not use repositories without TLS and trusted certificate in the build chain.

File: akka-bbb-fsesl/build.sbt [hunk 2]
```diff
@@ -28,8 +28,8 @@ val compileSettings = Seq(
 )
 
 resolvers ++= Seq(
-  "spray repo" at "http://repo.spray.io/",
-  "blindside-repos" at "http://blindside.googlecode.com/svn/repository/"
+  ("spray repo" at "http://repo.spray.io/").withAllowInsecureProtocol(true),
```
Comment (basisbit) [inline akka-bbb-fsesl/build.sbt:31]: - this is http only
- this only includes an "old" version of spray
- spray.io is not maintained any more for more than half a decade. Please migrate to akka http or similar. Also see https://doc.akka.io/docs/akka-http/current/migration-guide/migration-from-spray.html?language=scala

**Comment by basisbit** [comment]
any reason for not using the java version that comes by default with the maintained Ubuntu LTE repository?

File: bbb-common-message/project/Dependencies.scala [hunk 3]
```diff
@@ -7,21 +7,21 @@ object Dependencies {
...
-    val gson = "2.8.5"
-    val jackson = "2.9.7"
-    val sl4j = "1.7.25"
-    val red5 = "1.0.10-M9"
-    val pool = "2.8.0"
+    val akkaVersion = "2.6.17"
+    val gson = "2.8.9"
+    val jackson = "2.13.0"
+    val sl4j = "1.7.32"
+    val red5 = "1.2.10"
```
Comment (antobinary) [inline bbb-common-message/project/Dependencies.scala:20]: I believe this is obsolete too - we don't use red5 anymore

Comment (GhaziTriki) [reply bbb-common-message/project/Dependencies.scala:20]: True red5 can be safely removed.

Comment (himanshugoyal1065) [reply bbb-common-message/project/Dependencies.scala]: Taken care of!

**Comment by himanshugoyal1065** [comment]
> any reason for not using the java version that comes by default with the maintained Ubuntu LTE repository?

Hi @basisbit,
No specific reason. I was using sdkman and it was easy to switch between versions with that. 
I agree that it makes more sense to use the java LTE version provided by Ubuntu. I'll change that real quick soon.

Out of curiosity, how did you figure out that what version/provider of java I'm using?

**Comment by antobinary** [comment]
> Hey @antobinary, I tested this PR. After updating my local Gradle and Grails versions I ran some meetings and it seems like everything is working properly.

Awesome, @paultrudel ! Could you also do a code review on it too, please?
I know @GhaziTriki wants to do one too but this is such a significant change that it would benefit from extra scrutiny..

File: akka-bbb-apps/project/Dependencies.scala [hunk 4]
```diff
@@ -60,15 +60,15 @@ object Dependencies {
...
+//    val junit = "junit" % "junit" % Versions.junit % "test"
     val mockitoCore = "org.mockito" % "mockito-core" % Versions.mockito % "test"
     val scalactic = "org.scalactic" % "scalactic_2.12" % Versions.scalactic % "test"
     val akkaTestKit = "com.typesafe.akka" %% "akka-testkit" % Versions.akkaTestKit % "test"
   }
 
   val testing = Seq(
     Test.scalaTest,
-    Test.junit,
+//    Test.junit,
```
Comment (antobinary) [inline akka-bbb-apps/project/Dependencies.scala:71]: @paultrudel are you able to run the tests if you uncomment the 2 lines in this file?

File: akka-bbb-apps/project/build.properties [hunk 5]
```diff
@@ -1 +1 @@
-sbt.version=1.2.8
+sbt.version=1.5.5
```
Comment (antobinary) [inline akka-bbb-apps/project/build.properties:1]: Note to @antobinary -- we need to reflect this change in https://gitlab.senfcall.de/senfcall-public/docker-bbb-build/-/blob/main/Dockerfile#L135-137 in a new tag for 2.5

File: akka-bbb-fsesl/build.sbt [hunk 6]
```diff
@@ -79,4 +79,4 @@ daemonGroup in Linux := group
 
 javaOptions in Universal ++= Seq("-J-Xms130m", "-J-Xmx256m", "-Dconfig.file=/etc/bigbluebutton/bbb-fsesl-akka.conf", "-Dlogback.configurationFile=conf/logback.xml")
 
-debianPackageDependencies in Debian ++= Seq("java8-runtime-headless", "bash", "bbb-freeswitch-core")
+debianPackageDependencies in Debian ++= Seq("java17-runtime-headless", "bash", "bbb-freeswitch-core")
```
Comment (antobinary) [inline akka-bbb-fsesl/build.sbt:82]:
```suggestion
debianPackageDependencies in Debian ++= Seq("java11-runtime-headless", "bash", "bbb-freeswitch-core")
```
perhaps?

File: akka-bbb-fsesl/project/Dependencies.scala [hunk 7]
```diff
@@ -7,18 +7,18 @@ object Dependencies {
...
     val junitInterface = "0.11"
     val scalactic = "3.0.3"
 
     // Libraries
-    val akkaVersion = "2.5.19"
-    val akkaHttpVersion = "10.1.4"
-    val logback = "1.2.3"
+    val akkaVersion = "2.6.17"
+    val akkaHttpVersion = "10.2.7"
+    val logback = "1.3.0-alpha10"
```
Comment (antobinary) [inline akka-bbb-fsesl/project/Dependencies.scala:17]: `val logback = "1.3.0-alpha10"` <- is this production ready? Do we need anything specific that would justify an alpha? Is this related to the log4j security in any way?

File: bbb-lti/build.gradle [hunk 8]
```diff
@@ -126,3 +145,7 @@ assets {
   minifyJs = true
   minifyCss = true
 }
+
+springBoot {
+  mainClassName = 'org.bigbluebuttoni.Application'
```
Comment (antobinary) [inline bbb-lti/build.gradle:150]:
```suggestion
  mainClassName = 'org.bigbluebutton.Application'
```
Typo?

File: bbb-lti/gradlew [hunk 9]
```diff
@@ -1,25 +1,43 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
```
Comment (antobinary) [inline bbb-lti/gradlew:1]: I notice a file permission change 644 -> 755. Is this intentional?

File: bbb-lti/gradlew [hunk 10]
```diff
@@ -150,11 +154,19 @@ if $cygwin ; then
...
-eval splitJvmOpts $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS
-JVM_OPTS[${#JVM_OPTS[*]}]="-Dorg.gradle.appname=$APP_BASE_NAME"
+APP_ARGS=$(save "$@")
+
+# Collect all arguments for the java command, following the shell quoting and substitution rules
+eval set -- $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS "\"-Dorg.gradle.appname=$APP_BASE_NAME\"" -classpath "\"$CLASSPATH\"" org.gradle.wrapper.GradleWrapperMain "$APP_ARGS"
+
+# by default we should be in the correct project dir, but when run from Finder on Mac, the cwd is wrong
+if [ "$(uname)" = "Darwin" ] && [ "$HOME" = "$PWD" ]; then
+  cd "$(dirname "$0")"
```
Comment (antobinary) [inline bbb-lti/gradlew:169]: Does this have correct quotes? Seems like $0 might not be in quoted context

File: akka-bbb-apps/project/Dependencies.scala (hunk 4 above)
Comment (antobinary) [reply akka-bbb-apps/project/Dependencies.scala:71]: In direct conversation @paultrudel confirmed the tests are fine. Seems these two lines are commented out because they are not used.

File: akka-bbb-apps/build.sbt [hunk 11]
```diff
@@ -72,4 +72,4 @@ daemonGroup in Linux := group
 
 javaOptions in Universal ++= Seq("-J-Xms130m", "-J-Xmx256m", "-Dconfig.file=/etc/bigbluebutton/bbb-apps-akka.conf", "-Dlogback.configurationFile=conf/logback.xml")
 
-debianPackageDependencies in Debian ++= Seq("java8-runtime-headless", "bash")
+debianPackageDependencies in Debian ++= Seq("java17-runtime-headless", "bash")
```
[the remark this answers is not in the document]
Comment (himanshugoyal1065) [reply akka-bbb-apps/build.sbt:75]: Yes! I'll make this change. The PR started with upgrading the version to JDK 17. Hence this. 

Thanks for pointing it out.

File: akka-bbb-fsesl/project/Dependencies.scala (hunk 7 above)
Comment (himanshugoyal1065) [reply akka-bbb-fsesl/project/Dependencies.scala:17]: Yeah makes sense. 
I have email the logback team enquiring about the same question. For now, I have changed the version to [1.2.10] which is their latest stable release.

File: bbb-lti/build.gradle (hunk 8 above)
Comment (himanshugoyal1065) [reply bbb-lti/build.gradle:150]: I think this is how it should be considering the class package statement looks like this
`package org.bigbluebuttoni`

https://github.com/bigbluebutton/bigbluebutton/blob/8404048cfd35280d2dd427fc0c2e93790760c3f7/bbb-lti/grails-app/init/org/bigbluebutton/Application.groovy#L19

File: bbb-lti/gradlew (hunk 9 above)
Comment (himanshugoyal1065) [reply bbb-lti/gradlew:1]: I'm not sure about it because the file is auto-generated by gradle. This might has happened due to change in gradle version?

File: bbb-lti/gradlew (hunk 10 above)
Comment (himanshugoyal1065) [reply bbb-lti/gradlew:169]: Same comment as above. 

I'm not sure about it because the file is auto-generated by gradle. This might has happened due to change in gradle version?

**Comment by himanshugoyal1065** [comment]
> any reason for not using the java version that comes by default with the maintained Ubuntu LTE repository?

Hi @basisbit,
Noted and now I'm using ubuntu provided JDK 11. But still, both I and @antobinary are curious to know how did you get to know that I'm using some JDK17 apart from ububtu's.

File: akka-bbb-apps/project/Dependencies.scala [hunk 12]
```diff
@@ -7,23 +7,23 @@ object Dependencies {
...
-    val logback = "1.2.3"
-    val quicklens = "1.4.11"
-    val spray = "1.3.4"
+    val akkaVersion = "2.6.17"
+    val akkaHttpVersion = "10.2.7"
+    val gson = "2.8.9"
+    val jackson = "2.13.0"
+    val logback = "1.2.7"
+    val quicklens = "1.7.5"
+    val spray = "1.3.6"
```
Comment (antobinary) [inline akka-bbb-apps/project/Dependencies.scala:22]: Now that you removed `spray`, do we still need this variable?

**Comment by antobinary** [comment]
I pulled the code locally and tried to run. Here are some steps I had to do:

#### Setup before:

$ `java -version`
openjdk version "1.8.0_312"
OpenJDK Runtime Environment (build 1.8.0_312-8u312-b07-0ubuntu1~18.04-b07)
OpenJDK 64-Bit Server VM (build 25.312-b07, mixed mode)


$ `sudo update-alternatives --config java`

```
There is 1 choice for the alternative java (providing /usr/bin/java).

  Selection    Path                                            Priority   Status
------------------------------------------------------------
  0            /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java   1081      auto mode
* 1            /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java   1081      manual mode
```


$ `sdk current `

```
Using:

gradle: 5.5.1
grails: 3.3.9
maven: 3.5.0
sbt: 1.2.8
```

`echo "$JAVA_HOME"`
results in `/usr/lib/jvm/java-8-openjdk-amd64`

#### Changes:

$ `sudo apt-get install openjdk-11-jdk`

$ `sudo update-alternatives --config java`

```
There are 2 choices for the alternative java (providing /usr/bin/java).

  Selection    Path                                            Priority   Status
------------------------------------------------------------
  0            /usr/lib/jvm/java-11-openjdk-amd64/bin/java      1111      auto mode
  1            /usr/lib/jvm/java-11-openjdk-amd64/bin/java      1111      manual mode
* 2            /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java   1081      manual mode

Press <enter> to keep the current choice[*], or type selection number: 0
update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/java to provide /usr/bin/java (java) in auto mode
```

$ `java -version`
openjdk version "11.0.13" 2021-10-19
OpenJDK Runtime Environment (build 11.0.13+8-Ubuntu-0ubuntu1.18.04)
OpenJDK 64-Bit Server VM (build 11.0.13+8-Ubuntu-0ubuntu1.18.04, mixed mode, sharing)

`sdk install grails 5.0.1`  -- in the code I see you used this version
`sdk install sbt  1.5.5` --  in the code I see you used this version
~~sdk install maven 3.5.0~~ -- I did not see any info on maven change, so I skipped this


#### Interesting part

- When I tried to run bigbluebutton-web I hit an error, looks like we need to upgrade Gradle. I did not find info on which version you used. Based on the error I decided to proceed with `7.3.1`

<details>
bigbluebutton@anton25:~/src/bigbluebutton-web$ ./build.sh 
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Build file '/home/bigbluebutton/src/bigbluebutton-web/build.gradle' line: 24

* What went wrong:
A problem occurred evaluating root project 'bigbluebutton'.
> Failed to apply plugin [class 'org.springframework.boot.gradle.plugin.SpringBootPlugin']
   > Spring Boot plugin requires Gradle 6.8.x, 6.9.x, or 7.x. The current version is Gradle 5.5.1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 3s

FAILURE: Build failed with an exception.

* Where:
Build file '/home/bigbluebutton/src/bigbluebutton-web/build.gradle' line: 24

* What went wrong:
A problem occurred evaluating root project 'bigbluebutton'.
> Failed to apply plugin [class 'org.springframework.boot.gradle.plugin.SpringBootPlugin']
   > Spring Boot plugin requires Gradle 6.8.x, 6.9.x, or 7.x. The current version is Gradle 5.5.1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

| Resolving Dependencies. Please wait...
Downloading https://services.gradle.org/distributions/gradle-7.3.1-bin.zip
</details>

$ `sdk install gradle 7.3.1`

- I had to change JAVA_HOME
$ `vi ~/.profile`
set 
`export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64`

$ `source ~/.profile`
$ `echo "$JAVA_HOME"` results in `/usr/lib/jvm/java-11-openjdk-amd64`

- I noticed that in the `build.sbt` files you kept `"-target:jvm-1.8",`
I tried to change it to `"-target:jvm-1.11",` as I thought that was the essence of the effort.

I had to make a bunch of changes to get `bbb-common-message` to build, the rest of the components were mostly similar.
**I had to change sbt to 1.6.2 because that version allowed me to compile with scala 2.13.4, which seems to be one of the first fine with targeting JVM 11.** Otherwise sbt 1.5.5 was using scala 2.12 and I could not target JVM 11.

Here is a commit with my changes: https://github.com/antobinary/bigbluebutton/commit/4e268dffa34a7b0072c7b3f6730671918428b1c6
 With those I could build `bbb-common-message`, `bbb-web`, `akka-bbb-apps`, `bbb-fsesl-client`, `akka-bbb-fsesl`.

About `bigbluebutton-web` -- Looks like gradle 5.5 -> 7.3 was a huge jump. I am not sure if you used 7.3.1. Based on your changes you seem to be adapting `build.gradle` to handle the jump https://docs.gradle.org/current/userguide/upgrading_version_5.html

However, when I try to do `resolveDeps` it looks like it's not pulling any dependencies. Without them, the `./run.sh` step is missing beans from what I can tell...

Could you please comment on bbb-web building/running and on the JVM target?

**Comment by himanshugoyal1065** [comment]
Hi @antobinary 

Thanks for the review!

I must have missed the JVM target part and thanks for pointing it out. That upgrade seems good to me. 

Regarding scala version update from 2.12.x to 2.13.x, I was able to build the project with the following config on my server
`
root@ns382746:~/dev/bigbluebutton# sdk current
Using:
gradle: 7.3.1
grails: 5.0.1
maven: 3.8.4
sbt: 1.5.5
scala: 2.12.15
`
But, changing it to 2.3.x won't harm as the newer the better. 

But I'm not sure why weren't you able to build the project with `almost` the same configurations earlier. Maybe the version of `grails` caused that if you forgot to update by any chance?

Also, I pulled your changes on my setup and it worked perfectly fine. So, on mutual consensus (:P), we can go with the changes you added on top of mine.

**Comment by antobinary** [comment]
Because of https://github.com/bigbluebutton/bigbluebutton/pull/14489 we'll have to drop the changes to `bbb-lti`
